File: content/problems/2408-design-sql/solutions/per-table-append-only-row-slots/Solution.java
class SQL {
    Map<String,List<String[]>> tables=new HashMap<>();
    public SQL(String[] names,int[] columns){
        for(String name:names)tables.put(name,new ArrayList<>());
    }
    public void insertRow(String name,String[] row){
        tables.get(name).add(row.clone());
    }
    public void deleteRow(String name,int rowId){
        tables.get(name).set(rowId-1,null);
    }
    public String selectCell(String name,int rowId,int columnId){
        return tables.get(name).get(rowId-1)[columnId-1];
    }
}

File: content/problems/2408-design-sql/solutions/per-table-append-only-row-slots/Solution.kt
class SQL(names:Array<String>,columns:IntArray){
    private val tables=HashMap<String,MutableList<Array<String>?>>();
    init{
        for(name in names)tables[name]=mutableListOf()
    }
    fun insertRow(name:String,row:Array<String>){
        tables[name]!!.add(row.clone())
    }
    fun deleteRow(name:String,rowId:Int){
        tables[name]!![rowId-1]=null
    }
    fun selectCell(name:String,rowId:Int,columnId:Int):String=tables[name]!![rowId-1]!![columnId-1]
}
